test/error/require_fail.cpp

```diff
@@ -0,0 +1,24 @@
+#include "Halide.h"
+#include <stdio.h>
+#include <memory>
+
+using namespace Halide;
+
+int main(int argc, char **argv) {
+    const int kPrime1 = 7829;
+    const int kPrime2 = 7919;
+
+    Buffer<int> result;
+    Param<int> p1, p2;
+    Var x;
+    Func f;
+    f(x) = require((p1 + p2) == kPrime1, 
+                   (p1 + p2) * kPrime2,
+                   "The parameters should add to exactly", (kPrime1 * kPrime2), "but were", p1, p2);
```

Comment:
Not that it matters much, but I think the error should use kPrime1, or the condition should use the product.